### ISSUE TITLE
Preserve enclosing break ownership in all switch raisers

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3836]
## Problem

`SwitchRaisingPass.Raise` and `FinishSwitchRaise` can reparent an existing
`Break` from an enclosing loop into a newly raised `Switch`, changing which
construct the statement exits.

This was found while reviewing #3832, but it reproduces identically at base
`4ca46623` and is not caused or worsened by that PR. The case-target-join path
now guards every reparented region; the two older raisers do not apply the same
ownership rule.

## Reproduction

Release-compiled source:

```csharp
int result = 0;
do
{
    switch (value)
    {
        case 0:
            result += 1;
            break;
        case 1:
            result += 2;
            break;
        case 2:
            result += 3;
            break;
        case 3:
            result += 4;
            break;
        default:
            result += 100;
            goto Done;
    }
    result++;
}
while (result < 30);
Done:
return result;
```

The decompiled default becomes:

```csharp
default:
    V_0 += 100;
    break;
```

The original `goto Done` is raised by `DoWhileLoopPass` to a loop `Break`.
`SwitchRaisingPass.Raise` then moves that node into a switch section, where it
binds to the switch instead. For an unmatched value, original and decompiled
results differ (`100` versus `101`).

## Expected fix

Apply one shared enclosing-break ownership rule to every switch raiser that
reparents regions:

- `Raise`
- `FinishSwitchRaise`
- `RaiseCaseTargetJoin`

Cover direct and `IfStatement`-wrapped breaks, while retaining positive
coverage for breaks owned by region-local loops or switches. Use a
compiler-produced Release fixture in addition to hand-built IR.
